datafusion/physical-expr/src/equivalence.rs

```diff
@@ -198,36 +200,50 @@ impl OrderedColumn {
     }
 }
 
-trait ColumnAccessor {
-    fn column(&self) -> &Column;
-}
-
-impl ColumnAccessor for Column {
-    fn column(&self) -> &Column {
-        self
+impl From<OrderedColumn> for PhysicalSortExpr {
+    fn from(value: OrderedColumn) -> Self {
+        PhysicalSortExpr {
+            expr: Arc::new(value.col) as _,
+            options: value.options,
+        }
     }
 }
 
-impl ColumnAccessor for OrderedColumn {
-    fn column(&self) -> &Column {
-        &self.col
+impl From<OrderedColumn> for PhysicalSortRequirement {
+    fn from(value: OrderedColumn) -> Self {
+        PhysicalSortRequirement {
+            expr: Arc::new(value.col) as _,
+            options: Some(value.options),
+        }
     }
 }
 
-pub type OrderingEquivalentClass = EquivalentClass<OrderedColumn>;
+pub type OrderingEquivalentClass = EquivalentClass<Vec<OrderedColumn>>;
```

Comment:
I think it would help to add some doc comments here explaining that the `Vec` is the set of columns that is equivalent as well as mentioning what, if any, significance the ordering of the `OrderedColumn` in this type have